src/pushsource/_impl/model/ami.py

```diff
@@ -99,9 +99,42 @@ class AmiPushItem(VMIPushItem):
     )
     """Billing codes associated with this image."""
 
+    release_notes = attr.ib(type=str, default=None, validator=optional_str)
+    """"Notes regarding changes in the newest version. URLs are supported.
+    
+    SPECIFICATIONS:
+    * Open text field
+    * 1 character MIN
+    * 30000 characters MAX
+    """
+
+    usage_instructions = attr.ib(type=str, default=None, validator=optional_str)
+    """Step by step instructions for the end-user to launch, configure and access the product. 
+    Write with a less-technical customer in mind. Can include URL to more comprehensive steps.
+
+    SPECIFICATIONS:
+    * Open text field
+    * 4000 characters MAX
+    """
+
+    recommended_instance_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Recommended instance type for AMI e.g. "t2.micro", "r6a.large" """
+
+    marketplace_entity_type = attr.ib(type=str, default=None, validator=optional_str)
+    """Type of entity e.g. "AMIProduct", "SaasProduct", "ServerProduct". """
+
     image_id = attr.ib(type=str, default=None, validator=optional_str)
     """AMI Image ID used to reference the image in AWS."""
 
+    scanning_port = attr.ib(type=str, default=None, validator=optional_str)
+    """AMI scanning port, used when importing the AMI into AWS Marketplace to validate the AMI."""
```

Comment:
Could you give me some detail what this is please?  The reason is that it's declared as a string, but generally if we're talking about ports, `int` comes to mind.

Reply:
@jajreidy what do you think about this one?

Reply:
I just checked, that supposed to be int, my bad.

Reply:
Since the default is None, the validator has to allow for a value of None.  Right now it strictly requires int.

This is the reason for some tests failing with:

```
TypeError: ("'scanning_port' must be <class 'int'> (got None that is a <class 'NoneType'>)
```

Reply:
@ashwini3326 in this case, you can use `optional` as validator:

```
    scanning_port = attr.ib(type=int, default=None, validator=optional(instance_of(int)))
    """AMI scanning port, used when importing the AMI into AWS Marketplace to validate the AMI."""
```

Reply:
Is the new default of 0 valid on the AWS side?   How is it interpreted?  (Are there some docs about it?)

In general, 0 isn't a valid port number.  If that is the case here as well, it seems like the previous default of None was preferable.  Otherwise we're basically using two different values to indicate "no port number" which seems unnecessarily complex.

Reply:
Ashwini confirmed that 0 is accepted by the AWS APIs, so I think this is OK.  `None` means "we don't have a value for the field", 0 means whatever the AWS API does with 0.